cmd/main.go

```diff
@@ -108,10 +108,39 @@ func readYAMLFile() error {
 	if err := decoder.Decode(&baseline); err != nil {
 		return fmt.Errorf("error decoding YAML: %v", err)
 	}
+	var entryIDs []string
+	for i, entry := range baseline.Criteria {
+		// if entry in entryIDs
+		if contains(entryIDs, entry.ID) {
+			return fmt.Errorf("duplicate ID for criteria entry %d: %s", i, entry.ID)
+		}
+		if entry.ID == "" {
```

Comment:
Now that we have a naming pattern, we should add a regexp check here (we can follow up with that later)

Reply:
Ah I just saw this is the same comment @funnelfiasco made :)